### ISSUE TITLE
Feature request: Restrict zoom 

### ISSUE THREAD

**Issue by vassbence** [ValentinH/react-easy-crop#380]
I think it would be nice to have a feature where the minimum zoom value would be set automatically to such a number where the image is perfectly in the cropper. Meaning the user would not be able to scroll out so much that there is empty white space.

This image shows the current state:
<img width="877" alt="image" src="https://user-images.githubusercontent.com/49574140/166701712-d8aee86d-03b4-4a65-ab16-eeff965bda64.png">

And the expected state with this new feature:
<img width="865" alt="image" src="https://user-images.githubusercontent.com/49574140/166701883-5b83ab57-290e-48a3-99a5-6490dffba8da.png">

Could we simply calculate a minZoom value or this will need some more coding?

**Comment by ValentinH** [comment]
I'm not sure to understand because right now the default zoom is 1 which shows the whole image inside the cropper.
Could you share a sandbox showing the issue?

**Comment by vassbence** [comment]
If `vertical-cover` or `horizontal-cover` is used as `objectFit` and the minZoom is like 0.1 the user can zoom out so much that there is empty space in the cropper.

Sandbox: https://codesandbox.io/s/immutable-sun-7o438p
To recreate just zoom out

**Comment by ValentinH** [comment]
The thing is that even if we find the right minZoom, you will end up with white-space:
![image](https://user-images.githubusercontent.com/2678610/166781921-d35e3d6d-5a78-4800-b6b0-e8747b6986f6.png)

I played with different values and it turns out that the minZoom should be set to `crop_aspect / image_aspect`.  
Here's the updated sandbox: https://codesandbox.io/s/upbeat-boyd-7edv85

**Comment by vassbence** [comment]
I thought the math would need to include some sizes of the cropper (by cropper i mean the inner gridded area)

Can't seem to figure out the numbers either..

**Comment by ValentinH** [comment]
Doesn't my example match what you are looking for? I mean I term of minZoom value/behavior, even if it isn't dynamic yet?

**Comment by vassbence** [comment]
I might not have been clear, sorry about that, let me try do give you all the details about my use case:

We are doing it kinda weird, tall images get horizontal-cover while wide images get vertical-cover (`imageToCrop.width > imageToCrop.height * CROP_ASPECT`). Because of this most of the time the initial load with zoom 1 leads to a state where the user can actually zoom out while still getting a whole crop (so no white space).

Initial (zoom=1, vertical-cover, 1910x1000 pic): 
<img width="555" alt="image" src="https://user-images.githubusercontent.com/49574140/166840434-8848cc30-9208-48f8-9c41-b448c60bc677.png">


User zooming out from initial (zoom=~0.80): 
<img width="555" alt="image" src="https://user-images.githubusercontent.com/49574140/166840366-ced4d925-0960-4017-8295-e9bd61cb5f68.png">

---
So basically all I'm trying to do is to find the correct minZoom value which (in the example above its around 0.80 but it changes based on how wide the codesandbox window is) where the user can zoom out but only so much that the crop area is still fully in the image and there is no white space left.

My sandbox: https://codesandbox.io/s/stupefied-shape-bikdld?file=/src/index.js

---
With `crop_aspect / image_aspect` there is white space at the bottom when fully zoomed out.

**Comment by ValentinH** [comment]
I managed to find the right way for your specific example: https://codesandbox.io/s/recursing-antonelli-ud1x2e?file=/src/index.js

It has to be `container_aatio / crop_aspect` when the container is taller than wide. I guess this calculation is different for each `objectFit` value though and also depending on the container aspect.

**Comment by vassbence** [comment]
I figured it out with a bit of a hacky solution:

```
if (imageToCrop.width > imageToCrop.height * TODO_ASPECT)
	return;

const img = document.querySelector(".reactEasyCrop_Image");
const cropper = document.querySelector(
	".reactEasyCrop_CropArea",
);

if (img && cropper) {
	const imgDimensions = {
		width: img.offsetWidth,
		height: img.offsetHeight,
	};
	const cropperDimensions = cropper.getBoundingClientRect();

	const biggerSideOfCropper = Math.max(
		cropperDimensions.width,
		cropperDimensions.height,
	);
	const smallerSideOfImage = Math.min(
		imgDimensions.width,
		imgDimensions.height,
	);

	setMinZoom(biggerSideOfCropper / smallerSideOfImage);
}
```

But after thinking about it I don't think this is something that will be used by most people as we are doing a weird use case so I think the issue can be closed.

Thank you very much for helping me :) Lovely package.

**Comment by ValentinH** [comment]
I'm glad this works for you. Do you confirm that you can do this in your code without having to fork this package?

**Comment by vassbence** [comment]
Yup, can be done in userland without a fork!

**Comment by satyamskillz** [comment]
This is kinda dumb code but seems to work fine with all crop aspect ratios and image aspect ratios. 
```
import CrossSVG from "@/assets/Icons/CrossSVG";
import ModalLayout from "@/layouts/ModelLayout";

import { useElementSize } from "usehooks-ts";
import Cropper from "react-easy-crop";
import { useState } from "react";

function ImageCropper({ imageSrc, onClose, aspect = 16 / 9 }) {
	const [croppedAreaPixels, setCroppedAreaPixels] = useState(null);
	const [objectFit, setObjectFit] = useState("vertical-cover");
	const [crop, setCrop] = useState({ x: 0, y: 0 });
	const [zoom, setZoom] = useState(1);

	const [minZoom, setMinZoom] = useState(1);
	const [containerRef, { width = 0, height = 0 }] = useElementSize();

	const onCropComplete = (croppedArea, croppedAreaPixels) => {};

	const showCroppedImage = async () => {};

	const onMediaLoaded = (media) => {
		const ratio = media.width / (media.height * aspect);

		if (aspect === 1) {
			if (ratio < 1) {
				setObjectFit("horizontal-cover");
				setMinZoom(height / width);
			}
		} else if (aspect > 1) {
			if (ratio < 1) setObjectFit("horizontal-cover");
			if (ratio > 1) setMinZoom(width / height / aspect);
		}
	};

	return (
		<ModalLayout onClickOutside={onClose}>
			<div className="component__ImageCropper">
				<div className="head">
					<p className="title">Crop Image</p>
					<span onClick={onClose}>
						<CrossSVG />
					</span>
				</div>
				<div className="body" ref={containerRef}>
					<Cropper
						crop={crop}
						zoom={zoom}
						aspect={aspect}
						image={imageSrc}
						minZoom={minZoom}
						objectFit={objectFit}
						onCropChange={setCrop}
						onZoomChange={setZoom}
						restrictPosition={true}
						onMediaLoaded={onMediaLoaded}
						onCropComplete={onCropComplete}
					/>
				</div>
				<div className="foot">
					<button onClick={showCroppedImage}>
						Done
					</button>
				</div>
			</div>
		</ModalLayout>
	);
}

export default ImageCropper;
```
